### PR TITLE
Fix failed build reporting for Scala 2.11

### LINKED ISSUE #1120: Failing compilations succeed in Bazel
I am trying to bazelify SpinalHDL/VexRiscV, a sbt project, with the following rule:
```
load("@io_bazel_rules_scala//scala:scala.bzl", "scala_binary", "scala_library")

scala_library(
        name = "VexRiscV",
        srcs = glob(
            ["src/main/scala/**/*.scala"],
            exclude = ["src/main/scala/demo/**/*"]
        ),
        plugins = ["@maven//:com_github_spinalhdl_spinalhdl_idsl_plugin_2_11"],
        deps = [
            "@maven//:com_github_spinalhdl_spinalhdl_core_2_11",
            "@maven//:com_github_spinalhdl_spinalhdl_sim_2_11",
            "@maven//:com_github_spinalhdl_spinalhdl_lib_2_11",
            "@maven//:com_github_spinalhdl_spinalhdl_idsl_plugin_2_11",
            "@maven//:org_yaml_snakeyaml",
            "@maven//:commons_io_commons_io",
            "@io_bazel_rules_scala_scala_compiler//:io_bazel_rules_scala_scala_compiler",
            "@io_bazel_rules_scala_scala_library//:io_bazel_rules_scala_scala_library"
            ],
)
```
Attempting to build this rule results in a lot of errors (unrelated to this issue), but **bazel reports the build as successful**:
```INFO: Analyzed target //:VexRiscV (65 packages loaded, 1292 targets configured).
INFO: Found 1 target...
INFO: From scala //:VexRiscV:
src/main/scala/spinal/lib/bus/wishbone/Wishbone.scala:53: error: Symbol 'type spinal.idslplugin.ValCallback' is missing from the classpath.
This symbol is required by 'trait spinal.core.ValCallbackRec'.
Make sure that type ValCallback is in your classpath and check for conflicting dependencies with `-Ylog-classpath`.
A full rebuild may help if 'ValCallbackRec.class' was compiled against an incompatible version of spinal.idslplugin.

...
many more errors here
...
src/main/scala/vexriscv/plugin/MmuPlugin.scala:254: error: type mismatch;
 found   : spinal.core.spinal.core.Area
 required: spinal.core.spinal.core.Area
    fenceStage plug new Area{
                    ^
Target //:VexRiscV up-to-date:
  bazel-bin/VexRiscV.jar
INFO: Elapsed time: 11.593s, Critical Path: 10.27s
INFO: 2 processes: 2 worker.
INFO: Build completed successfully, 7 total actions
```
The jar is created empty - contains just a MANIFEST.
From my basic bisecting it appears this was introduced in 2aaa7bcc104d40e.

### PR DESCRIPTION
**Pull request by liucijus**
fixes #1120

### CODE REVIEW

File: src/java/io/bazel/rulesscala/scalac/ReportableMainClass.java [hunk 1]
```diff
@@ -4,41 +4,44 @@
...
 
-    public ReportableMainClass(CompileOptions ops){
+    public ReportableMainClass(CompileOptions ops) {
         this.ops = ops;
     }
 
     @Override
     public Global newCompiler() {
+        if (!ops.enableDiagnosticsReport) {
+            createDiagnosticsFile();
```
Comment (simuons) [inline src/java/io/bazel/rulesscala/scalac/ReportableMainClass.java:23]: This looks a bit confusing to me. Diagnostics file is created even when diagnostics is not enabled.
Diagnostics file is created in two branches but doesn't seem to be used. Maybe it's a wrong place to create this file?

Comment (liucijus) [reply src/java/io/bazel/rulesscala/scalac/ReportableMainClass.java:23]: Diagnostics is still defined as rule output, so it expects this file to be produced - this is how original implementation works.

**Comment by liucijus** [comment]
To clarify important bits regarding this change:
- this fixes broken 2.11 support - I think it's important to get this fixed ASAP
- to do it properly, diagnostics reporter should be fixed also when it is enabled for all supported versions of Scala. This means we need to have failing test scenarios for all major Scala versions for error reporting. Unfortunately I have no time to look into it myself. Any chances @andrefmrocha you can look into it?

**Comment by andrefmrocha** [comment]
> to do it properly, diagnostics reporter should be fixed also when it is enabled for all supported versions of Scala. This means we need to have failing test scenarios for all major Scala versions for error reporting. Unfortunately I have no time to look into it myself. Any chances @andrefmrocha you can look into it?

Thanks for looking into this!

I unfortunately wasn't able to look into this earlier due to hardware issues that did not allow me to work on this. However, this hardware issues should be fixed by the start of next week, and I'll gladly look into this.

Just to make sure I understood, your changes guarantee that compilation fails whenever the diagnostics reporter is not used, meaning that it will still break when the diagnostics reporter is used. Is this correct?

**Comment by liucijus** [comment]
> Just to make sure I understood, your changes guarantee that compilation fails whenever the diagnostics reporter is not used, meaning that it will still break when the diagnostics reporter is used. Is this correct?

Yes, I think so. I only short circuited to the old reporting mechanism when diagnostics reporter is off.

**Comment by ittaiz** [comment]
I hate merging without a test but maybe we should make an exception given @andrefmrocha is committed to add tests.
What I'm still not sure is why your fix fixes anything. Do you know?

**Comment by liucijus** [comment]
> I hate merging without a test but maybe we should make an exception given @andrefmrocha is committed to add tests.
> What I'm still not sure is why your fix fixes anything. Do you know?

**Comment by ittaiz** [comment]
ok, does that answer the question?

**Pull request merged**